slack/web/async_client.py

```diff
@@ -2156,3 +2174,154 @@ async def views_publish(
+        """Invite a user to a public or private channel.
+
+        Args:
+            channel_id (str): The channel that the users will be invited to.
+            user_ids (str or list): The users to invite.
+        """
+        kwargs.update({"channel_id": channel_id})
+        if isinstance(user_ids, list):
+            kwargs.update({"user_ids": ",".join(user_ids)})
+        else:
+            kwargs.update({"user_ids": user_ids})
+        return await self.api_call("admin.conversations.invite", json=kwargs)
+
+    async def admin_conversations_archive(
+        self, channel_id: str, **kwargs
+    ) -> AsyncSlackResponse:
+        """Archive a public or private channel.
+
+        Args:
+            channel_id (str): The channel to archive.
+        """
+        kwargs.update({"channel_id": channel_id})
+        return await self.api_call("admin.conversations.archive", json=kwargs)
+
+    async def admin_conversations_unarchive(
+        self, channel_id: str, **kwargs
+    ) -> AsyncSlackResponse:
+        """Unarchive a public or private channel.
+
+        Args:
+            channel_id (str): The channel to unarchive.
+        """
+        kwargs.update({"channel_id": channel_id})
+        return await self.api_call("admin.conversations.unarchive", json=kwargs)
+
+    async def admin_conversations_rename(
+        self, channel_id: str, name: str, **kwargs
+    ) -> AsyncSlackResponse:
+        """Rename a public or private channel.
+
+        Args:
+            channel_id (str): The channel to rename.
+            name (str): The name to rename the channel to.
+        """
+        kwargs.update({"channel_id": channel_id, "name": name})
+        return await self.api_call("admin.conversations.rename", json=kwargs)
+
+    async def admin_conversations_search(self, **kwargs) -> AsyncSlackResponse:
+        """Search for public or private channels in an Enterprise organization."""
+        return await self.api_call("admin.conversations.search", json=kwargs)
+
+    async def admin_conversations_convertToPrivate(
+        self, channel_id: str, **kwargs
+    ) -> AsyncSlackResponse:
+        """Convert a public channel to a private channel.
+
+        Args:
+            channel_id (str): The channel to convert to private.
+        """
+        kwargs.update({"channel_id": channel_id})
+        return await self.api_call("admin.conversations.convertToPrivate", json=kwargs)
+
+    async def admin_conversations_setConversationPrefs(
+        self, channel_id: str, prefs: Union[str, dict], **kwargs
+    ) -> AsyncSlackResponse:
+        """Set the posting permissions for a public or private channel.
+
+        Args:
+            channel_id (str): The channel to set the prefs for
+            prefs (str or dict): The prefs for this channel in a stringified JSON format.
+        """
+        kwargs.update({"channel_id": channel_id})
+        if isinstance(prefs, dict):
+            kwargs.update({"prefs": dumps(prefs)})
```

Comment:
I don't think you have to do this here. `BaseClient` handles the serialization even for nested structure.